lib/OpenQA/Schema/Result/Needles.pm

```diff
@@ -203,15 +189,9 @@ sub remove {
     return 0;
 }
 
-sub check_file {
-    my ($self) = @_;
-
-    return $self->file_present(-e $self->path ? 1 : 0);
-}
-
-sub to_json {
-    my ($self, $controller) = @_;
+sub check_file ($self) { $self->file_present(-e $self->path ? 1 : 0) }
 
+sub to_json ($self, $controller) {
```

Comment:
Looks like `$controller` isn't used. Maybe it makes more sense to removed (removing it from the call site as well).

Reply:
Yeah, this looks like an accident. It is never called with a second argument, so just remove it.

Reply:
done